lib/gauche/regexp.scm

```diff
@@ -34,7 +34,11 @@
 ;; NB: refrain from using util.match, for it could cause nasty dependency
 ;; problem.  (Should be resolved once we incorporate match into the core).
```

Comment:
This "nasty dependency problem" comment scares me. I have no clue if pulling more imports here would cause problems.

Reply:
Yes, it can be tricky.  A general rule is that gauche.* module shouldn't depend on anything other than other gauche.* modules.  I know I violate it occasionally.  But the closer to the "core" the module is, the more careful we should be.  gauche.regexp is very close to the core; I wouldn't let it depend on scheme.* modules.

My suggestion is to start implementing sre in separate module---let's say gauche.regexp.sre.  We can set autoload in gauche.regexp as well so that it looks like the API is exported from gauche.regexp.  But in gauche.regexp.sre you can depend on whatever modules, for simply using gauche.regexp (without sre API) won't load gauche.regexp.sre.   After we finish implementation and testing, we can examine the dependency closely and may be able to integrate gauche.regexp.sre into gauche.regexp.